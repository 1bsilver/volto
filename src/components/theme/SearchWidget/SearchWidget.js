/**
 * Search widget component.
 * @module components/theme/SearchWidget/SearchWidget
 */

import React from 'react';
import { browserHistory } from 'react-router';

/**
 * SearchWidget component class.
 * @function SearchWidget
 * @returns {string} Markup of the component.
 */
const SearchWidget = () => (
  <div id="portal-searchbox">
    <form id="searchGadget_form" action="" role="search" className="pat-livesearch" onSubmit={e => {
      browserHistory.push(`/search?SearchableText=${document.getElementById('searchGadget').value}`);
      e.preventDefault();
      return false;
    }}>
      <div className="LSBox">
        <label className="hiddenStructure" htmlFor="searchGadget">Search Site</label>

        <input name="SearchableText" type="text" size="18" id="searchGadget" title="Search Site" placeholder="Search Site" className="searchField" autoComplete="off" /><ul className="livesearch-results" style={{ display: 'none' }}></ul>

        <input className="searchButton" type="submit" value="Search" />

        <div className="searchSection">
          <input id="searchbox_currentfolder_only" className="noborder" type="checkbox" name="path" value="/Plone18" />
          <label htmlFor="searchbox_currentfolder_only" style={{ cursor: 'pointer' }}>only in current section</label>
        </div>
      </div>
    </form>
  </div>
);

export default SearchWidget;
