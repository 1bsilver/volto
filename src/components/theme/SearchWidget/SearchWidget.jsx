/**
 * Search widget component.
 * @module components/theme/SearchWidget/SearchWidget
 */

import React from 'react';
import { browserHistory } from 'react-router';
import { Checkbox, Form, Input } from 'semantic-ui-react';

/**
 * SearchWidget component class.
 * @function SearchWidget
 * @returns {string} Markup of the component.
 */
const SearchWidget = () => (
  <Form
    action="/search"
    onSubmit={(e) => {
      browserHistory.push(`/search?SearchableText=${document.getElementsByName('SearchableText')[0].value}`);
      e.preventDefault();
      return false;
    }}
  >
    <Form.Field>
      <Input name="SearchableText" action="Search" placeholder="Search Site" />
    </Form.Field>
    <Form.Field>
      <Checkbox label='only in current section' />
    </Form.Field>
  </Form>
);

export default SearchWidget;
