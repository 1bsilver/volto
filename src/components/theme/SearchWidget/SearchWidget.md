Search Widget, with `intl` auto-injected:

```jsx static
<SearchWidget pathname="" />
```

Output:

```jsx_ noeditor
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;
import {StaticRouter} from 'react-router-dom';
const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <SearchWidget pathname="" />
    </StaticRouter>
  </Provider>
</div>;
```
