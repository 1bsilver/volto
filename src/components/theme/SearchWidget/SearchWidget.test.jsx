import React from 'react';
import renderer from 'react-test-renderer';
import SearchWidget from './SearchWidget';

test('renders a not found component', () => {
  const component = renderer.create(<SearchWidget />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
