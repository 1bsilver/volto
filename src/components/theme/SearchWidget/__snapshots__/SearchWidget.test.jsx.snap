// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchWidget renders a search widget component 1`] = `
<form
  action="/search"
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="ui action input"
    >
      <input
        name="SearchableText"
        onChange={[Function]}
        placeholder="Search Site"
        type="text"
        value=""
      />
      <button
        className="ui button button"
        onClick={[Function]}
        tabIndex={undefined}
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value={undefined}
      />
      <label>
        only in current section
      </label>
    </div>
  </div>
</form>
`;
